src/analysis/souffle/check_predicate.dl

```diff
@@ -24,6 +24,7 @@
 
 .type CheckPredicate =
   CP_TagPresence { ap: AccessPath, principal: Principal, tag: Tag }
+  | CP_TagPresenceDynamicOwner { ap: AccessPath, tag: Tag }
```

Comment:
I see why you introduced this, but it actually weakens the check. See the comment around `tagPresentForSomeOwner`.

Reply:
See new version of this, I think I've addressed the issue.